### PR TITLE
Testing honza dudek

### PR DESCRIPTION
**Pull request by HonzaDudek**
Fixed test files for: buildings, login, logout, register, app, game components and services.

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[0111/131256.178:ERROR:prune_crash_reports.cc(34)] PruneCrashReportDatabase: Failed to get pending reports
```
Comment (gyopak) [inline debug.log:1]: ignore this file

File: src/app/_helpers/interceptors/buildings.interceptor.ts [hunk 2]
```diff
@@ -18,24 +18,20 @@ export class BuildingsInterceptor implements HttpInterceptor {
...
 
     if (req.url.endsWith('/game/buildings')) {
-      return new Observable(observer => observer.next(new HttpResponse<any>({
-          body: {
-            buildings,
-          },
-          status: 200,
-        })
-      ));
+      console.log('/game/buildings was called');
```
Comment (gyopak) [inline src/app/_helpers/interceptors/buildings.interceptor.ts]: console.log

File: src/app/_helpers/interceptors/buildings.interceptor.ts [hunk 3]
```diff
@@ -18,24 +18,20 @@ export class BuildingsInterceptor implements HttpInterceptor {
...
+      console.log(utilities.sendResponse(buildings, 200));
+      return utilities.sendResponse({buildings}, 200);
     } else if (req.url.endsWith(environment.createBuilding)) {
-      let newBuilding = buildingsMock.createBuilding(req.body.type);
-      return utilities.sendResponse({type: newBuilding.type}, 200);
+        let newBuilding = buildingsMock.createBuilding(req.body.type);
```
Comment (gyopak) [inline src/app/_helpers/interceptors/buildings.interceptor.ts]: should be `const`

File: src/app/_helpers/interceptors/buildings.interceptor.ts [hunk 4]
```diff
@@ -18,24 +18,20 @@ export class BuildingsInterceptor implements HttpInterceptor {
...
+        return utilities.sendResponse({type: newBuilding.type}, 200);
     } else if (req.url.endsWith(`/game/buildings/${req.body.id}`) && req.method === 'PUT') {
-      let buildingToUpdate = buildingsMock.findBuilding(req.body.id);
-      buildingToUpdate.level += 1;
-      buildingsMock.updateBuilding(buildingToUpdate);
-      return utilities.sendResponse({}, 200);
+        let buildingToUpdate = buildingsMock.findBuilding(req.body.id);
```
Comment (gyopak) [inline src/app/_helpers/interceptors/buildings.interceptor.ts]: should be `const`

File: src/app/_helpers/interceptors/buildings.interceptor.ts [hunk 5]
```diff
@@ -18,24 +18,20 @@ export class BuildingsInterceptor implements HttpInterceptor {
...
+        buildingToUpdate.level += 1;
+        buildingsMock.updateBuilding(buildingToUpdate);
+        return utilities.sendResponse({}, 200);
     } else if (req.url.endsWith('/game/buildings/' + req.body.id) && req.method === 'GET') {
```
Comment (gyopak) [inline src/app/_helpers/interceptors/buildings.interceptor.ts]: also this is a crazy if-else if- else if combination, let's try to refactor it a bit

File: src/app/_helpers/interceptors/buildings.interceptor.ts [hunk 6]
```diff
@@ -18,24 +18,20 @@ export class BuildingsInterceptor implements HttpInterceptor {
...
-      let building = buildingsMock.findBuilding(req.body.id);
-      return utilities.sendResponse({building}, 200);
+        let building = buildingsMock.findBuilding(req.body.id);
```
Comment (gyopak) [inline src/app/_helpers/interceptors/buildings.interceptor.ts]: `const`

File: src/app/alert/alert.component.spec.ts [hunk 7]
```diff
@@ -1,17 +1,57 @@
...
+import {LogoutComponent} from '../logout/logout.component';
+import {RegisterComponent} from '../register/register.component';
+import {HeaderComponent} from '../header/header.component';
+import {GameComponent} from '../game/game.component';
+import {ResourcesComponent} from '../game/resources/resources.component';
 import {RouterTestingModule} from '@angular/router/testing';
-import {Router} from '@angular/router';
+import {BrowserModule} from '@angular/platform-browser';
+import {AppRoutingModule} from '../app.routes';
+import {HttpModule} from '@angular/http';
```
Comment (gyopak) [inline src/app/alert/alert.component.spec.ts]: you don't use it

File: src/app/game/buildings/building-details/building-detail.service.spec.ts [hunk 8]
```diff
@@ -1,12 +1,79 @@
...
-    const service: BuildingDetailService = TestBed.get(BuildingDetailService);
     expect(service).toBeTruthy();
   });
+
+  it('should level up building on click', async () => {
+    spyOn(buildingDetailComponent, 'levelUpBuilding');
+
+    let buttons = fixture.debugElement.queryAll(By.css('button'));
+    console.log(buttons);
+    buttons[1].triggerEventHandler('click', buildingDetailComponent);
```
Comment (gyopak) [inline src/app/game/buildings/building-details/building-detail.service.spec.ts]: this button will be the first every time? query it with class name or smth

File: src/app/game/buildings/buildings.service.spec.ts [hunk 9]
```diff
@@ -1,21 +1,62 @@
...
   it('should be created', () => {
-    const service: BuildingsService = TestBed.get(BuildingsService);
     expect(service).toBeTruthy();
   });
+
+  it('should return list of buildings from the database', () => {
+    let buildings: Building[];
+    service.getBuildings().subscribe(response => {
+      buildings.push(response.body);
+      expect(buildings[0]).toEqual(
```
Comment (gyopak) [inline src/app/game/buildings/buildings.service.spec.ts:38]: a bit hacky test but ok

File: src/app/logout/logout.service.spec.ts [hunk 10]
```diff
@@ -1,73 +1,114 @@
...
+    });
+
+    httpClient = TestBed.get(HttpClient);
+    logoutService = TestBed.get(LogoutService);
+    router = TestBed.get(Router);
+    httpTestingController = TestBed.get(HttpTestingController);
+
+    let store = {};
+    const mockLocalStorage = {
+      getItem: (key: string): string => {
```
Comment (gyopak) [inline src/app/logout/logout.service.spec.ts:75]: `getItem: (key: string): string => store[key] || null` could work also

File: src/app/_helpers/interceptors/buildings.interceptor.ts (hunk 2 above)
Comment (HonzaDudek) [reply src/app/_helpers/interceptors/buildings.interceptor.ts]: deleted

File: src/app/game/buildings/building-details/building-detail.service.spec.ts (hunk 8 above)
Comment (HonzaDudek) [reply src/app/game/buildings/building-details/building-detail.service.spec.ts]: Changed to class selector

**Review by gyopak — APPROVED** [review]

**Pull request merged**